supernode/p2p/service.go

```diff
@@ -0,0 +1,70 @@
+package p2p
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"github.com/pastelnetwork/gonode/common/log"
+)
+
+const (
+	logPrefix = "p2p"
+)
+
+// Service represents the p2p service.
+type Service struct {
+	dht DHT
+}
+
+// DHT represents the methods by which a library consumer interacts with a DHT.
+type DHT interface {
+	Store(data []byte, key []byte) (id string, err error)
+	Get(key string) (data []byte, found bool, err error)
+	CreateSocket() error
+	Listen() error
```

Comment:
```suggestion
	Listen(ctx context.Context) error
```
Yeah, you have to improve their code, I quickly found an [example](https://www.reddit.com/r/golang/comments/jk5sv6/how_to_gracefully_stop_a_tcp_server/), maybe there is a better example